### PR TITLE
Return Empty Result With Empty Nonempty Domain With Multi-Index

### PR DESCRIPTION
**Pull request by nguyenv**
* Previously, this operation would error out.

### CODE REVIEW

File: tiledb/tests/test_multi_index.py [hunk 1]
```diff
@@ -760,3 +758,20 @@ def test_multi_index_with_implicit_full_string_range(self):
...
+        schema = tiledb.ArraySchema(domain=dom, sparse=True, attrs=[attr])
+        tiledb.Array.create(uri, schema)
+
+        with tiledb.open(uri, mode="w", timestamp=2) as A:
+            d1 = np.array(np.random.randint(1, 11, size=3, dtype=np.int32))
+            A[np.arange(1, 4)] = d1
+
+        with tiledb.open(uri, mode="r", timestamp=1) as A:
+            assert A.nonempty_domain() is None
+            assert_array_equal(A.multi_index[:][""], A[:][""])
```
Comment (ihnorton) [inline tiledb/tests/test_multi_index.py:775]: What does this return value look like?

Comment (nguyenv) [reply tiledb/tests/test_multi_index.py:775]: It is an empty list `[]`

Comment (ihnorton) [reply tiledb/tests/test_multi_index.py:775]: Thanks

**Review by ihnorton — APPROVED** [review]
Thanks!

**Pull request merged**